mitodl/mit-open
# Preliminary ETL pipeline for Sloan

### What are the relevant tickets?
Closes https://github.com/mitodl/hq/issues/4732

Related PR: https://github.com/mitodl/ol-infrastructure/pull/2625

### Description (What does it do?)
- Adds a new ETL pipeline to ingest courses and programs directly from Sloan's own API feed instead of Prolearn's.
- Excludes Sloan courses from being ingested by the Prolearn ETL pipeline.


### How can this be tested?
- Set the following values in your .env file:
  ```
   PROLEARN_CATALOG_API_URL=https://prolearn.mit.edu/graphql
   SEE_API_ENABLED=True
   SEE_API_URL
   SEE_API_ACCESS_TOKEN_URL
   SEE_API_CLIENT_ID
   SEE_API_CLIENT_SECRET
  ```
-  On the **main** branch, run `./manage.py backpopulate_prolearn_data`
-  When it is done, there should be learning resources with `etl_source=prolearn` and `offered_by=see`.  They will all be courses and not programs.  http://api.open.odl.local:8063/admin/learning_resources/learningresource/?o=2&offered_by__code__exact=see
- If you don't already have a populated featured learning path for Sloan, run `./manage.py populate_featured_lists` to create one

- Switch to this branch.  Run `./manage.py backpopulate_sloan_data` followed by `./manage.py backpopulate_prolearn_data`)  
- If you go to `http://open.odl.local:8062/learningpaths` and open the Professional Education featured learning path, it will be empty.  Run the following to reassign the old featured courses from prolearn to. their new equivalents
```
./manage.py transfer_list_resources course title prolearn see
```
- Once complete, you should have courses in the Sloan featured learning path again.  These commands will need to be run on RC/production as well to retain the current featured lists.

To test the `SEE_API_ENABLED` flag:
- Remove it from your backend env file or set it to `False`
- Run `./manage.py backpopulate_sloan_data`, it should not ingest any courses
- Run `./manage.py backpopulate_prolearn_data`, it should ingest (and set to published) Sloan courses
